test/features/test_multiscale_feature_persistence.cpp

```diff
@@ -0,0 +1,191 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2021-, Open Perception
+ *
+ *  All rights reserved
+ */
+
+#include <pcl/common/norms.h>
+#include <pcl/features/fpfh.h>
+#include <pcl/features/multiscale_feature_persistence.h>
+#include <pcl/test/gtest.h>
+#include <pcl/point_cloud.h>
+
+using PointT = pcl::PointXYZ;
+using PointCloudT = pcl::PointCloud<PointT>;
+using FPFHSignature = pcl::FPFHSignature33;
+PointCloudT::Ptr cloud;
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+TEST(PCL, DistanceBetweenFeatures)
+{
+  pcl::detail::MultiscaleFeaturePersistenceTest<PointT, FPFHSignature> test_distance;
+  std::vector<float> a{101, -11, 24, 3, 18, 27, 65};
+  std::vector<float> b{89, 29, 24, 1008, -57, 106, 85};
+
+  // L1
+
+  test_distance.distance_metric_ = pcl::L1;
+  EXPECT_EQ(test_distance.distanceBetweenFeatures(a, b), pcl::L1_Norm(a, b, a.size()));
+
+  // L2_SQR
+
+  test_distance.distance_metric_ = pcl::L2_SQR;
+  EXPECT_EQ(test_distance.distanceBetweenFeatures(a, b),
+            pcl::L2_Norm_SQR(a, b, a.size()));
+
+  // L2
+  test_distance.distance_metric_ = pcl::L2;
+  EXPECT_EQ(test_distance.distanceBetweenFeatures(a, b), pcl::L2_Norm(a, b, a.size()));
+
+  // LINF
+  test_distance.distance_metric_ = pcl::LINF;
+  EXPECT_EQ(test_distance.distanceBetweenFeatures(a, b),
+            pcl::Linf_Norm(a, b, a.size()));
+
+  // JM
+  // test_distance.distance_metric_ = pcl::JM;
+  // EXPECT_EQ(test_distance.distanceBetweenFeatures(a, b), pcl::JM_Norm(a, b,
+  // a.size()));
+
+  // B
+  test_distance.distance_metric_ = pcl::B;
+  EXPECT_EQ(test_distance.distanceBetweenFeatures(a, b), pcl::B_Norm(a, b, a.size()));
+
+  // SUBLINEAR
+  test_distance.distance_metric_ = pcl::SUBLINEAR;
+  EXPECT_EQ(test_distance.distanceBetweenFeatures(a, b),
+            pcl::Sublinear_Norm(a, b, a.size()));
+
+  // CS
+  test_distance.distance_metric_ = pcl::CS;
+  EXPECT_EQ(test_distance.distanceBetweenFeatures(a, b), pcl::CS_Norm(a, b, a.size()));
+
+  // DIV
+  test_distance.distance_metric_ = pcl::DIV;
+  EXPECT_EQ(test_distance.distanceBetweenFeatures(a, b), pcl::Div_Norm(a, b, a.size()));
+
+  // KL
+  test_distance.distance_metric_ = pcl::KL;
+  EXPECT_EQ(test_distance.distanceBetweenFeatures(a, b), pcl::KL_Norm(a, b, a.size()));
+
+  // HIK
+  test_distance.distance_metric_ = pcl::HIK;
+  EXPECT_EQ(test_distance.distanceBetweenFeatures(a, b), pcl::HIK_Norm(a, b, a.size()));
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST(PCL, CalculateMeanFeature)
+{
+  pcl::detail::MultiscaleFeaturePersistenceTest<PointT, FPFHSignature> test_mean;
+  test_mean.features_at_scale_vectorized_ = {{{1, 2, 3, 4, 5}, {6, 2, 9, 4, 21}},
```

Comment:
Same comment regarding the values. `1-5`, `11-15`, `21-25`, etc. might be good choice, with some negatives groups of numbers sprinkled in between